packages/server/graphql/queries/helpers/resolveSelectedTemplate.ts

```diff
@@ -2,22 +2,40 @@ import getRethink from '../../../database/rethinkDriver'
 import MeetingSettingsPoker from '../../../database/types/MeetingSettingsPoker'
 import MeetingSettingsRetrospective from '../../../database/types/MeetingSettingsRetrospective'
 import {GQLContext} from '../../graphql'
+import {getUserId} from '../../../utils/authorization'
+import isValid from '../../isValid'
 
 const resolveSelectedTemplate =
   (fallbackTemplateId: string) =>
   async (
     source: MeetingSettingsPoker | MeetingSettingsRetrospective,
     _args: unknown,
-    {dataLoader}: GQLContext
+    {authToken, dataLoader}: GQLContext
   ) => {
+    const viewerId = getUserId(authToken)
     const {id: settingsId, selectedTemplateId, teamId} = source
-    const [team, template] = await Promise.all([
+    const [team, template, viewer] = await Promise.all([
       dataLoader.get('teams').loadNonNull(teamId),
-      dataLoader.get('meetingTemplates').load(selectedTemplateId)
+      dataLoader.get('meetingTemplates').load(selectedTemplateId),
+      dataLoader.get('users').loadNonNull(viewerId)
     ])
     const {tier} = team
-    if (template?.isFree || template?.scope !== 'PUBLIC' || tier !== 'starter') {
-      return template
+    if (template) {
+      if (
+        template.isFree ||
+        template.scope !== 'PUBLIC' ||
+        tier !== 'starter' ||
+        viewer.featureFlags.includes('noTemplateLimit')
+      ) {
+        return template
+      }
+      // if anyone on the team has the noTemplateLimit flag, they might have selected a non-starter template
```

Comment:
maybe easier to have org level feature flag in this case

Reply:
Yes, that's true. I think we discussed this at some point, but I cannot remember the argument. I just went with the implementation issue. I'll merge it as is, but might rework this in a separate PR if we change our mind.